MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -287,6 +287,13 @@ - (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo fromUserNotificati
         [self.delegate push:self didReceivePushNotification:pushNotification];
       });
 #if TARGET_OS_OSX
+    } else {
+      NSUserNotification *notification = [[NSUserNotification alloc] init];
+      notification.title = title;
+      notification.informativeText = message;
+      notification.userInfo = userInfo;
+      NSUserNotificationCenter *center = [NSUserNotificationCenter defaultUserNotificationCenter];
+      [center deliverNotification:notification];
     }
 #endif
```

Comment:
I have a feedback on readability of existing code, not directly on your new code in this PR. I found [Line 271](https://github.com/Microsoft/mobile-center-sdk-ios/pull/715/files?diff=unified#diff-8a6dda2e220279e67126fd356f3f5381R271) to [Line 298](https://github.com/Microsoft/mobile-center-sdk-ios/pull/715/files?diff=unified#diff-8a6dda2e220279e67126fd356f3f5381R298) are not slightly less readable than I expected because of nested compiler directives `#if TARGET_OS_OSX` inside `if-else` conditions. To be more specific, two pairs of `#if TARGET_OS_OSX` `#endif` makes it hard to understand the logic. To improve readability I hope there could be some refactoring here. At least you can refactor out the shared logic between different platforms to avoid this.